build-scripts/build_rule_playbooks.py

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/python3
+
+import argparse
+import os
+import re
+import sys
+
+from collections import OrderedDict
+
+import ssg.rules
+import ssg.utils
+import ssg.yaml
+import ssg.build_yaml
+import ssg.build_remediations
+
+COMMENTS_TO_PARSE = ["strategy", "complexity", "disruption"]
+
+
+class PlaybookBuilder():
+    def __init__(self, build_config_yaml, product_yaml, input_dir, output_dir):
+        self.input_dir = input_dir
+        self.output_dir = output_dir
+        self.env_yaml = ssg.yaml.open_environment(build_config_yaml,
+                                                  product_yaml)
+        product_dir = os.path.dirname(product_yaml)
+        relative_guide_dir = ssg.utils.required_key(self.env_yaml,
+                                                    "benchmark_root")
+        self.guide_dir = os.path.abspath(os.path.join(product_dir,
+                                                      relative_guide_dir))
+        relative_profiles_dir = ssg.utils.required_key(self.env_yaml,
+                                                       "profiles_root")
+        self.profiles_dir = os.path.abspath(
+            os.path.join(product_dir, relative_profiles_dir))
+
+    def get_profile_selections(self, profile):
+        """
+        Provides a tuple (rules, variables) where rules is a list of rules
+        selected in the profile and variables is a dictionary of
+        variables and their values in the profile. The method can handle
+        profiles which extend other profiles.
+        """
+        rules = []
+        variables = dict()
+        if profile.extends:
+            extended_profile_path = os.path.join(
+                self.profiles_dir, profile.extends + ".profile")
+            extended_profile = ssg.build_yaml.Profile.from_yaml(
+                extended_profile_path, self.env_yaml)
+            if not profile:
+                sys.stderr.write(
+                    "Could not parse profile %s.\n" % extended_profile_path)
+                return None
+            rules, variables = self.get_profile_selections(extended_profile)
```

Comment:
I have a feeling that processing of "extened rules" should also be part of "Profile" class, but not sure if it will be useful.
I'm fine if it remains here though.

Reply:
The profile class would need to have context of other profiles. I think it isn't necessary for now.